test/test_magnet.cpp

```diff
@@ -279,6 +279,20 @@ TORRENT_TEST(parse_space_hash)
 	ec.clear();
 }
 
+TORRENT_TEST(parse_peer)
+{
+	error_code ec;
+	add_torrent_params p;
+	parse_magnet_uri("magnet:?xt=urn:btih:cdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcd&dn=foo&x.pe=127.0.0.1:43&x.pe=<invalid1>&x.pe=<invalid2>:100&x.pe=[::1]:45", p, ec);
+	TEST_CHECK(!ec);
+	TEST_EQUAL(p.peers.size(), 2);
+	TEST_EQUAL(p.peers[0].address(), address::from_string("127.0.0.1"));
```

Comment:
this could be shorter (and I think more readable) by using the convenience function to create endpoints, ep() https://github.com/arvidn/libtorrent/blob/master/test/setup_transfer.hpp#L109